### ISSUE TITLE
TimeoutOverflowWarning when using invalid config values

### ISSUE THREAD

**Issue by asherbig** [cross-seed/cross-seed#704]
### Have you checked for any existing open/closed tickets that are the same or similar to your issue?

- [X] I've searched far and wide!

### I have gone to the Discord server for support and/or acknowledge that this is a bug and not a support issue

- [X] I've tried to solve this already

### What's happening surrounding the bug? (context)

When setting delay to 3,000,000,000, then running `cross-seed search`, you get the error: `(node:23) TimeoutOverflowWarning: 3000000000 does not fit into a 32-bit signed integer.`

### Reproducing the bug

Set the delay in the config to `3_000_000_000`

### What should be happening?

This should fail at the config validation step (when launching the container). This seems to actually fail when the delay value is accessed, which happens once per search call.

Moreso, when this fails, the default value for delay is set to 1, which is an extremely dangerous value to use. If the value is overflowed, and you do _not_ wish to fail validation, then I would suggest setting the value to the max 32-bit integer value, and probably still printing a warning during the config validation step.

### Your setup

```markdown
- OS: Unraid
- Prowlarr: (not relevant)
- Docker
```

### config.js (REDACT SENSITIVE INFORMATION)

```
{
  ...
  delay: 3_000_000_000,
  ...
}
```

### verbose logs

```
info: Validating your configuration...
info: Your configuration is valid!
info: [search] Found 1384 torrents, 1040 suitable to search for matches
info: [1/1040] Searching for ---
info: [torznab] Searching 1 indexers for ---
(node:23) TimeoutOverflowWarning: 3000000000 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
(Use `node --trace-warnings ...` to show where the warning was created)
info: [2/1040] Searching for ---
info: [torznab] Searching 1 indexers for ---
(node:23) TimeoutOverflowWarning: 3000000000 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
info: [3/1040] Searching for ---
info: [torznab] Searching 1 indexers for ---
(node:23) TimeoutOverflowWarning: 3000000000 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
```

### Failure or refusal to provide the information we require to address bugs will result in potential closure of the issue.

- [X] I acknowledge that i have searched existing issues, read the documentation and faq, read the logs to see if the answer was provided in error messaging, and attempted to get support via the Discord prior to submitting this issue.

**Comment by zakkarry** [comment]
Not trying to dismiss a bug, but in what circumstances would a delay be set over ~60 years to provoke this?

It's a relatively easy problem to solve  I just also think it's rather far-fetched that it would ever be a problem given the conditions required to produce it.

Did you hunt this down or was there a reason you set it to over 100 years initially?

**Comment by zakkarry** [comment]
I opened a PR to fix this, after it is merged you will see an error during config validation.
![image](https://github.com/cross-seed/cross-seed/assets/123845855/89f0630e-b3bc-4856-a9bc-d9ac85637041)

**Comment by asherbig** [comment]
> Not trying to dismiss a bug, but in what circumstances would a delay be set over ~60 years to provoke this?

I just wanted a really long delay (not a realistic delay, just something so high, I didn't have to worry about it), and ran into this

**Comment by zakkarry** [comment]
> > Not trying to dismiss a bug, but in what circumstances would a delay be set over ~60 years to provoke this?
> 
> I just wanted a really long delay (not a realistic delay, just something so high, I didn't have to worry about it), and ran into this

I understand that a "set it and forget it" type of setup is ideal, typically I recommend something between 30 and 40 seconds. Even the most restrictive trackers API rate limits (150/hr) are only ~25s delay. 30-40 provides room for searches you are doing yourself, RSS updates, etc....

You literally would not have lived to see the second search with a delay of 3 billion lol.

I've fixed this, as its obviously not intended behavior, and I checked to make sure nothing else would produce this bug as well.

I suggest you look into searchLimit, excludeRecentSearch, excludeOlder, and the searchCadence options to even out the load on your indexers as you perform searches.

Delaying each search by over 100 years is not the way :P
